### PR TITLE
Poroelastic frac propagation criterion: use total stress to compute force balance.

### PR DESCRIPTION
**Pull request by CusiniM**
https://github.com/GEOS-DEV/integratedTests/pull/82

### CODE REVIEW

**Comment by binw1** [comment]
Tested and validated that total stress is used for nodal force calculation.

**Comment by paveltomin** [comment]
> Tested and validated that total stress is used for nodal force calculation.

Please approve it, somehow it got closed instead

**Review by binw1 — APPROVED** [review]
Tested and validated that total stress is used for nodal force calculation w/ penny shaped models.

File: src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp [hunk 1]
```diff
@@ -2893,322 +2907,313 @@ void SurfaceGenerator::calculateNodeAndFaceSif( DomainPartition const & domain,
     displacement.move( hostMemorySpace, false );
   } );
 
-  for( localIndex const trailingFaceIndex : m_trailingFaces )
-//  RAJA::forall< parallelHostPolicy >( RAJA::TypedRangeSegment< localIndex >( 0, m_trailingFaces.size() ),
-//                                      [=] GEOS_HOST_DEVICE ( localIndex const trailingFacesCounter )
+  // auto nodalForceKernel = surfaceGenerationKernels::createKernel( elementManager, constitutiveManager,
```
Comment (paveltomin) [inline src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp:2911]: remove or make RAJA?

File: src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp [hunk 2]
```diff
@@ -2893,322 +2907,313 @@ void SurfaceGenerator::calculateNodeAndFaceSif( DomainPartition const & domain,
...
+
+  surfaceGenerationKernels::kernelSelector( elementManager,
+                                            constitutiveManager,
+                                            viewKeyStruct::solidMaterialNameString(),
+                                            m_isPoroelastic, [&] ( auto nodalForceKernel )
   {
-//    localIndex const trailingFaceIndex = m_trailingFaces[ trailingFacesCounter ];
+    for( localIndex const trailingFaceIndex : m_trailingFaces )
+    {
+      //  RAJA::forall< parallelHostPolicy >( RAJA::TypedRangeSegment< localIndex >( 0, m_trailingFaces.size() ), [=] GEOS_HOST_DEVICE (
```
Comment (paveltomin) [inline src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp:2921]: same here

File: src/coreComponents/physicsSolvers/surfaceGeneration/kernels/surfaceGenerationKernels.hpp [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  }
+  else
+  {
+    string const porosityModelKey = constitutive::CoupledSolidBase::viewKeyStruct::porosityModelNameString();
+    lambda( PoroElasticNodalForceKernel( elemManager, constitutiveManager, solidMaterialKey, porosityModelKey )  );
+  }
+
+}
+
+// Alternative if we want to avoid another lambda.
```
Comment (paveltomin) [inline src/coreComponents/physicsSolvers/surfaceGeneration/kernels/surfaceGenerationKernels.hpp]: keep or remove?

**Comment by paveltomin** [comment]
@CusiniM can we merge it so that we can start testing QN without making two-branch combinations?

**Review by paveltomin — APPROVED** [review]
please merge

**Comment by paveltomin** [comment]
please-please merge

File: src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp (hunk 2 above)
Comment (CusiniM) [reply src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp:2921]: I don't think these have ever been raja loops tbh. We won't be ale to execute them on device so I don't know if it's worth changing it to a raja loop. Makes the syntax heavier.

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp:2921]: ok, sounds good

**Pull request merged**